{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene(s)</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>Condition(s)</th>\n",
       "      <th>Review status</th>\n",
       "      <th>Accession</th>\n",
       "      <th>GRCh37Chromosome</th>\n",
       "      <th>GRCh37Location</th>\n",
       "      <th>GRCh38Chromosome</th>\n",
       "      <th>GRCh38Location</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>AlleleID(s)</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <th>Unnamed: 15</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_000257.4(MYH7):c.816A&gt;T (p.Arg272Ser)</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>R272S</td>\n",
       "      <td>not provided</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>VCV002663533</td>\n",
       "      <td>14</td>\n",
       "      <td>23900189</td>\n",
       "      <td>14</td>\n",
       "      <td>23430980</td>\n",
       "      <td>2663533</td>\n",
       "      <td>2830981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000014.9:23430979:T:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_000257.4(MYH7):c.1072C&gt;T (p.His358Tyr)</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>H358Y</td>\n",
       "      <td>not provided</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>VCV002663260</td>\n",
       "      <td>14</td>\n",
       "      <td>23899050</td>\n",
       "      <td>14</td>\n",
       "      <td>23429841</td>\n",
       "      <td>2663260</td>\n",
       "      <td>2830710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000014.9:23429840:G:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NM_000257.4(MYH7):c.196G&gt;A (p.Gly66Ser)</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>G66S</td>\n",
       "      <td>not provided</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>VCV002662427</td>\n",
       "      <td>14</td>\n",
       "      <td>23902746</td>\n",
       "      <td>14</td>\n",
       "      <td>23433537</td>\n",
       "      <td>2662427</td>\n",
       "      <td>2829878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000014.9:23433536:C:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_000257.4(MYH7):c.639+1G&gt;T</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not provided</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>VCV002644107</td>\n",
       "      <td>14</td>\n",
       "      <td>23900969</td>\n",
       "      <td>14</td>\n",
       "      <td>23431760</td>\n",
       "      <td>2644107</td>\n",
       "      <td>2810291</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000014.9:23431759:C:A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_000257.4(MYH7):c.1669C&gt;A (p.Leu557Met)</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>L557M</td>\n",
       "      <td>not provided</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>VCV002644106</td>\n",
       "      <td>14</td>\n",
       "      <td>23897013</td>\n",
       "      <td>14</td>\n",
       "      <td>23427804</td>\n",
       "      <td>2644106</td>\n",
       "      <td>2810290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000014.9:23427803:G:T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Uncertain significance</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Name Gene(s) Protein change  \\\n",
       "0   NM_000257.4(MYH7):c.816A>T (p.Arg272Ser)    MYH7          R272S   \n",
       "1  NM_000257.4(MYH7):c.1072C>T (p.His358Tyr)    MYH7          H358Y   \n",
       "2    NM_000257.4(MYH7):c.196G>A (p.Gly66Ser)    MYH7           G66S   \n",
       "3               NM_000257.4(MYH7):c.639+1G>T    MYH7            NaN   \n",
       "4  NM_000257.4(MYH7):c.1669C>A (p.Leu557Met)    MYH7          L557M   \n",
       "\n",
       "   Condition(s)                        Review status     Accession  \\\n",
       "0  not provided  criteria provided, single submitter  VCV002663533   \n",
       "1  not provided  criteria provided, single submitter  VCV002663260   \n",
       "2  not provided  criteria provided, single submitter  VCV002662427   \n",
       "3  not provided  criteria provided, single submitter  VCV002644107   \n",
       "4  not provided  criteria provided, single submitter  VCV002644106   \n",
       "\n",
       "  GRCh37Chromosome GRCh37Location GRCh38Chromosome GRCh38Location  \\\n",
       "0               14       23900189               14       23430980   \n",
       "1               14       23899050               14       23429841   \n",
       "2               14       23902746               14       23433537   \n",
       "3               14       23900969               14       23431760   \n",
       "4               14       23897013               14       23427804   \n",
       "\n",
       "   VariationID AlleleID(s) dbSNP ID            Canonical SPDI  Unnamed: 15  \\\n",
       "0      2663533     2830981      NaN  NC_000014.9:23430979:T:A          NaN   \n",
       "1      2663260     2830710      NaN  NC_000014.9:23429840:G:A          NaN   \n",
       "2      2662427     2829878      NaN  NC_000014.9:23433536:C:T          NaN   \n",
       "3      2644107     2810291      NaN  NC_000014.9:23431759:C:A          NaN   \n",
       "4      2644106     2810290      NaN  NC_000014.9:23427803:G:T          NaN   \n",
       "\n",
       "                    label  \n",
       "0  Uncertain significance  \n",
       "1  Uncertain significance  \n",
       "2  Uncertain significance  \n",
       "3  Uncertain significance  \n",
       "4  Uncertain significance  "
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Downloaded only the relevant results for each gene from CLinVar, may have to change depeending on results\n",
    "\n",
    "cv_mybpc3_df = pd.read_csv(\"data/mybpc3_data.txt\", delimiter=\"\\t\")\n",
    "cv_myh7_df = pd.read_csv(\"data/myh7_data.txt\", delimiter=\"\\t\")\n",
    "\n",
    "#Removed last reviewed date from the disease status\n",
    "cv_myh7_df[\"label\"] = cv_myh7_df[\"Clinical significance (Last reviewed)\"].apply(lambda x: x.split(\"(\")[0])\n",
    "cv_myh7_df.drop(columns=[\"Clinical significance (Last reviewed)\"], inplace=True)\n",
    "\n",
    "cv_mybpc3_df[\"label\"] = cv_mybpc3_df[\"Clinical significance (Last reviewed)\"].apply(lambda x: x.split(\"(\")[0])\n",
    "cv_mybpc3_df.drop(columns=[\"Clinical significance (Last reviewed)\"], inplace=True)\n",
    "\n",
    "\n",
    "cv_mybpc3_df.head()\n",
    "cv_myh7_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>genome</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>am_pathogenicity</th>\n",
       "      <th>am_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23412857</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1935D</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23412857</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1935D</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23412858</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1935A</td>\n",
       "      <td>0.2034</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23412858</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1935G</td>\n",
       "      <td>0.1999</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23412858</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1935V</td>\n",
       "      <td>0.2693</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS REF ALT genome uniprot_id      transcript_id  \\\n",
       "0  chr14  23412857   C   G   hg38     P12883  ENST00000355349.4   \n",
       "1  chr14  23412857   C   A   hg38     P12883  ENST00000355349.4   \n",
       "2  chr14  23412858   T   G   hg38     P12883  ENST00000355349.4   \n",
       "3  chr14  23412858   T   C   hg38     P12883  ENST00000355349.4   \n",
       "4  chr14  23412858   T   A   hg38     P12883  ENST00000355349.4   \n",
       "\n",
       "  protein_variant  am_pathogenicity       am_class  \n",
       "0          E1935D            0.0737  likely_benign  \n",
       "1          E1935D            0.0737  likely_benign  \n",
       "2          E1935A            0.2034  likely_benign  \n",
       "3          E1935G            0.1999  likely_benign  \n",
       "4          E1935V            0.2693  likely_benign  "
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load AlphaMissense predictions for the two genes\n",
    "\n",
    "am_myh7_df = pd.read_csv(\"data/am_myh7_data.tsv\", delimiter=\"\\t\")\n",
    "am_mybpc3_df = pd.read_csv(\"data/am_mybpc3_data.tsv\", delimiter=\"\\t\")\n",
    "\n",
    "am_myh7_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wt_aa</th>\n",
       "      <th>position</th>\n",
       "      <th>mt_aa</th>\n",
       "      <th>ClinVar_ClinicalSignificance</th>\n",
       "      <th>Gold_Stars</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Starred_Coarse_Grained_Clin_Sig</th>\n",
       "      <th>frequency_gv2</th>\n",
       "      <th>frequency_gv3</th>\n",
       "      <th>evolutionary_index_ASM</th>\n",
       "      <th>...</th>\n",
       "      <th>p_model</th>\n",
       "      <th>b_acmg_model</th>\n",
       "      <th>lb_acmg_model</th>\n",
       "      <th>lp_acmg_model</th>\n",
       "      <th>coarse_clinical_significance_post</th>\n",
       "      <th>clinical_significance_post</th>\n",
       "      <th>model_disagreement</th>\n",
       "      <th>CV_label_and_model_combined_with_other_evidence_disagreement</th>\n",
       "      <th>starred_label_and_model_disagreement</th>\n",
       "      <th>starred_label_and_model_combined_with_other_evidence_disagreement</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  wt_aa  position mt_aa ClinVar_ClinicalSignificance  Gold_Stars  \\\n",
       "0     M         1     A                          NaN         NaN   \n",
       "1     M         1     C                          NaN         NaN   \n",
       "2     M         1     D                          NaN         NaN   \n",
       "3     M         1     E                          NaN         NaN   \n",
       "4     M         1     F                          NaN         NaN   \n",
       "\n",
       "   NumberSubmitters Starred_Coarse_Grained_Clin_Sig  frequency_gv2  \\\n",
       "0               NaN                             NaN            NaN   \n",
       "1               NaN                             NaN            NaN   \n",
       "2               NaN                             NaN            NaN   \n",
       "3               NaN                             NaN            NaN   \n",
       "4               NaN                             NaN            NaN   \n",
       "\n",
       "   frequency_gv3  evolutionary_index_ASM  ...  p_model  b_acmg_model  \\\n",
       "0            NaN                     NaN  ...    False         False   \n",
       "1            NaN                     NaN  ...    False         False   \n",
       "2            NaN                     NaN  ...    False         False   \n",
       "3            NaN                     NaN  ...    False         False   \n",
       "4            NaN                     NaN  ...    False         False   \n",
       "\n",
       "  lb_acmg_model lp_acmg_model coarse_clinical_significance_post  \\\n",
       "0         False         False                               NaN   \n",
       "1         False         False                               NaN   \n",
       "2         False         False                               NaN   \n",
       "3         False         False                               NaN   \n",
       "4         False         False                               NaN   \n",
       "\n",
       "  clinical_significance_post model_disagreement  \\\n",
       "0                        NaN              False   \n",
       "1                        NaN              False   \n",
       "2                        NaN              False   \n",
       "3                        NaN              False   \n",
       "4                        NaN              False   \n",
       "\n",
       "  CV_label_and_model_combined_with_other_evidence_disagreement  \\\n",
       "0                                              False             \n",
       "1                                              False             \n",
       "2                                              False             \n",
       "3                                              False             \n",
       "4                                              False             \n",
       "\n",
       "  starred_label_and_model_disagreement  \\\n",
       "0                                False   \n",
       "1                                False   \n",
       "2                                False   \n",
       "3                                False   \n",
       "4                                False   \n",
       "\n",
       "  starred_label_and_model_combined_with_other_evidence_disagreement  \n",
       "0                                              False                 \n",
       "1                                              False                 \n",
       "2                                              False                 \n",
       "3                                              False                 \n",
       "4                                              False                 \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Load EVE predictions for the two genes\n",
    "\n",
    "eve_myh7_df = pd.read_csv(\"data/eve_myh7_data.csv\")\n",
    "eve_mybpc3_df = pd.read_csv(\"data/eve_mybpc3_data.csv\")\n",
    "\n",
    "eve_myh7_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1279\n",
      "1279\n",
      "870\n",
      "870\n"
     ]
    }
   ],
   "source": [
    "#Preprocess ClinVar data\n",
    "\n",
    "for cv_df in [cv_myh7_df, cv_mybpc3_df]:\n",
    "    drop_idx_conflicting = cv_df[(cv_df[\"label\"] == 'Conflicting interpretations of pathogenicity') | (cv_df[\"label\"] == 'conflicting data from submitters') | (cv_df[\"label\"] == 'not provided')].index\n",
    "    drop_idx_unprov = cv_df[cv_df[\"label\"] == 'not provided'].index\n",
    "\n",
    "    #Normalize ClinVar labels to match those of AM\n",
    "    cv_df.drop(drop_idx_conflicting, inplace=True)\n",
    "\n",
    "    print(len(cv_df[cv_df[\"label\"].str.contains(\"Benign\") | cv_df[\"label\"].str.contains(\"benign\")]))\n",
    "    cv_df[\"label\"] = cv_df[\"label\"].apply(lambda x: \"ambiguous\" if x == \"Uncertain significance\" else (\"likely_benign\" if \"Benign\" in x or \"benign\" in x else \"likely_pathogenic\"))\n",
    "\n",
    "    print(len(cv_df.loc[cv_df[\"label\"] == \"likely_benign\"]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n",
      "/tmp/ipykernel_6011/414588944.py:27: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_am_pred)\n"
     ]
    }
   ],
   "source": [
    "#Find AM discrepancies with ClinVar for both genes\n",
    "\n",
    "mismatch_dict = {}\n",
    "for cv_df, am_df, gene in [(cv_myh7_df, am_myh7_df, \"am_myh7\"), (cv_mybpc3_df, am_mybpc3_df, \"am_mybpc3\")]:\n",
    "\n",
    "    mismatch_df = pd.DataFrame()\n",
    "    print()\n",
    "    for i in range(len(cv_df)):\n",
    "        matched_am_pred = am_df[am_df[\"protein_variant\"] == cv_df.iloc[i][\"Protein change\"]]\n",
    "\n",
    "\n",
    "        if not matched_am_pred.empty:\n",
    "            matched_am_pred = matched_am_pred.iloc[0]\n",
    "\n",
    "            exp_class = cv_df.iloc[i][\"label\"]\n",
    "            status = cv_df.iloc[i][\"Review status\"]\n",
    "            pred_class = matched_am_pred[\"am_class\"]\n",
    "\n",
    "            if exp_class != \"ambiguous\":\n",
    "                if exp_class != pred_class:\n",
    "\n",
    "                    matched_am_pred[\"ClinVar label\"] = exp_class\n",
    "                    matched_am_pred[\"ClinVar status\"] = status\n",
    "\n",
    "                    # print(matched_am_pred)\n",
    "                    # print(\"flow\")\n",
    "\n",
    "                    mismatch_df = mismatch_df.append(matched_am_pred)\n",
    "    \n",
    "    mismatch_dict[gene] = mismatch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "61\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>genome</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>am_pathogenicity</th>\n",
       "      <th>am_class</th>\n",
       "      <th>ClinVar label</th>\n",
       "      <th>ClinVar status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8260</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23426061</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>V689L</td>\n",
       "      <td>0.9478</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11544</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23431801</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>A200V</td>\n",
       "      <td>0.5483</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8838</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23427672</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>L601V</td>\n",
       "      <td>0.4057</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5822</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23422262</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>L1055M</td>\n",
       "      <td>0.1496</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10066</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23429095</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>A423T</td>\n",
       "      <td>0.1865</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5917</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23422306</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>Q1040R</td>\n",
       "      <td>0.2753</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7008</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424818</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M877T</td>\n",
       "      <td>0.2905</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23414026</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>K1879R</td>\n",
       "      <td>0.2235</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8508</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23427241</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R652K</td>\n",
       "      <td>0.5207</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5979</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23423556</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>Q1030H</td>\n",
       "      <td>0.3505</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7238</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424921</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>A843T</td>\n",
       "      <td>0.3338</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7964</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23425781</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>Q734E</td>\n",
       "      <td>0.2651</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10763</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23430607</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>T318P</td>\n",
       "      <td>0.4731</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3688</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23418244</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>A1379S</td>\n",
       "      <td>0.3965</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7382</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424984</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M822V</td>\n",
       "      <td>0.2755</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23413832</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>A1906G</td>\n",
       "      <td>0.3791</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7379</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424983</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M822T</td>\n",
       "      <td>0.4654</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7172</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424893</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M852R</td>\n",
       "      <td>0.4344</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8451</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23426842</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>T660N</td>\n",
       "      <td>0.1591</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7630</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23425358</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R783G</td>\n",
       "      <td>0.2878</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7173</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424893</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M852K</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6689</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424058</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E924G</td>\n",
       "      <td>0.4713</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8574</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23427274</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>G641A</td>\n",
       "      <td>0.5164</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7246</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424924</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>S842G</td>\n",
       "      <td>0.2516</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7272</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424938</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>K837R</td>\n",
       "      <td>0.2118</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8782</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23427647</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>Y609C</td>\n",
       "      <td>0.5154</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9960</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23429047</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M439V</td>\n",
       "      <td>0.4803</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6883</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>D896N</td>\n",
       "      <td>0.4739</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10247</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23429304</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>D394E</td>\n",
       "      <td>0.2842</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6664</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424047</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>D928N</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4911</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23419993</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R1193H</td>\n",
       "      <td>0.4463</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7202</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424905</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E848G</td>\n",
       "      <td>0.5390</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6614</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424026</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E935K</td>\n",
       "      <td>0.5593</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6524</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23423984</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E949K</td>\n",
       "      <td>0.5559</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7195</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424902</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M849T</td>\n",
       "      <td>0.5624</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7085</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424854</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>K865R</td>\n",
       "      <td>0.1168</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7089</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424855</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>K865E</td>\n",
       "      <td>0.4285</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11538</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23431798</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>I201T</td>\n",
       "      <td>0.2086</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6979</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424804</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>Q882E</td>\n",
       "      <td>0.3260</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7174</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424893</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>M852T</td>\n",
       "      <td>0.3544</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7061</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424843</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R869C</td>\n",
       "      <td>0.3800</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6825</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424118</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R904H</td>\n",
       "      <td>0.4928</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6650</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424041</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E930Q</td>\n",
       "      <td>0.4718</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23413845</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E1902Q</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7212</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424909</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>K847E</td>\n",
       "      <td>0.4053</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6896</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424148</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>E894G</td>\n",
       "      <td>0.3963</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6802</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424107</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>L908V</td>\n",
       "      <td>0.4203</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7947</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23425774</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>I736T</td>\n",
       "      <td>0.3767</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4017</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23418398</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>N1327K</td>\n",
       "      <td>0.8678</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7055</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23424840</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>P12883</td>\n",
       "      <td>ENST00000355349.4</td>\n",
       "      <td>R870C</td>\n",
       "      <td>0.3476</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>reviewed by expert panel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS REF ALT genome uniprot_id      transcript_id  \\\n",
       "8260   chr14  23426061   C   G   hg38     P12883  ENST00000355349.4   \n",
       "11544  chr14  23431801   G   A   hg38     P12883  ENST00000355349.4   \n",
       "8838   chr14  23427672   G   C   hg38     P12883  ENST00000355349.4   \n",
       "5822   chr14  23422262   G   T   hg38     P12883  ENST00000355349.4   \n",
       "10066  chr14  23429095   C   T   hg38     P12883  ENST00000355349.4   \n",
       "5917   chr14  23422306   T   C   hg38     P12883  ENST00000355349.4   \n",
       "7008   chr14  23424818   A   G   hg38     P12883  ENST00000355349.4   \n",
       "375    chr14  23414026   T   C   hg38     P12883  ENST00000355349.4   \n",
       "8508   chr14  23427241   C   T   hg38     P12883  ENST00000355349.4   \n",
       "5979   chr14  23423556   T   G   hg38     P12883  ENST00000355349.4   \n",
       "7238   chr14  23424921   C   T   hg38     P12883  ENST00000355349.4   \n",
       "7964   chr14  23425781   G   C   hg38     P12883  ENST00000355349.4   \n",
       "10763  chr14  23430607   T   G   hg38     P12883  ENST00000355349.4   \n",
       "3688   chr14  23418244   C   A   hg38     P12883  ENST00000355349.4   \n",
       "7382   chr14  23424984   T   C   hg38     P12883  ENST00000355349.4   \n",
       "193    chr14  23413832   G   C   hg38     P12883  ENST00000355349.4   \n",
       "7379   chr14  23424983   A   G   hg38     P12883  ENST00000355349.4   \n",
       "7172   chr14  23424893   A   C   hg38     P12883  ENST00000355349.4   \n",
       "8451   chr14  23426842   G   T   hg38     P12883  ENST00000355349.4   \n",
       "7630   chr14  23425358   G   C   hg38     P12883  ENST00000355349.4   \n",
       "7173   chr14  23424893   A   T   hg38     P12883  ENST00000355349.4   \n",
       "6689   chr14  23424058   T   C   hg38     P12883  ENST00000355349.4   \n",
       "8574   chr14  23427274   C   G   hg38     P12883  ENST00000355349.4   \n",
       "7246   chr14  23424924   T   C   hg38     P12883  ENST00000355349.4   \n",
       "7272   chr14  23424938   T   C   hg38     P12883  ENST00000355349.4   \n",
       "8782   chr14  23427647   T   C   hg38     P12883  ENST00000355349.4   \n",
       "9960   chr14  23429047   T   C   hg38     P12883  ENST00000355349.4   \n",
       "6883   chr14  23424143   C   T   hg38     P12883  ENST00000355349.4   \n",
       "10247  chr14  23429304   G   T   hg38     P12883  ENST00000355349.4   \n",
       "6664   chr14  23424047   C   T   hg38     P12883  ENST00000355349.4   \n",
       "4911   chr14  23419993   C   T   hg38     P12883  ENST00000355349.4   \n",
       "7202   chr14  23424905   T   C   hg38     P12883  ENST00000355349.4   \n",
       "6614   chr14  23424026   C   T   hg38     P12883  ENST00000355349.4   \n",
       "6524   chr14  23423984   C   T   hg38     P12883  ENST00000355349.4   \n",
       "7195   chr14  23424902   A   G   hg38     P12883  ENST00000355349.4   \n",
       "7085   chr14  23424854   T   C   hg38     P12883  ENST00000355349.4   \n",
       "7089   chr14  23424855   T   C   hg38     P12883  ENST00000355349.4   \n",
       "11538  chr14  23431798   A   G   hg38     P12883  ENST00000355349.4   \n",
       "6979   chr14  23424804   G   C   hg38     P12883  ENST00000355349.4   \n",
       "7174   chr14  23424893   A   G   hg38     P12883  ENST00000355349.4   \n",
       "7061   chr14  23424843   G   A   hg38     P12883  ENST00000355349.4   \n",
       "6825   chr14  23424118   C   T   hg38     P12883  ENST00000355349.4   \n",
       "6650   chr14  23424041   C   G   hg38     P12883  ENST00000355349.4   \n",
       "223    chr14  23413845   C   G   hg38     P12883  ENST00000355349.4   \n",
       "7212   chr14  23424909   T   C   hg38     P12883  ENST00000355349.4   \n",
       "6896   chr14  23424148   T   C   hg38     P12883  ENST00000355349.4   \n",
       "6802   chr14  23424107   G   C   hg38     P12883  ENST00000355349.4   \n",
       "7947   chr14  23425774   A   G   hg38     P12883  ENST00000355349.4   \n",
       "4017   chr14  23418398   G   T   hg38     P12883  ENST00000355349.4   \n",
       "7055   chr14  23424840   G   A   hg38     P12883  ENST00000355349.4   \n",
       "\n",
       "      protein_variant  am_pathogenicity           am_class      ClinVar label  \\\n",
       "8260            V689L            0.9478  likely_pathogenic      likely_benign   \n",
       "11544           A200V            0.5483          ambiguous  likely_pathogenic   \n",
       "8838            L601V            0.4057          ambiguous  likely_pathogenic   \n",
       "5822           L1055M            0.1496      likely_benign  likely_pathogenic   \n",
       "10066           A423T            0.1865      likely_benign  likely_pathogenic   \n",
       "5917           Q1040R            0.2753      likely_benign  likely_pathogenic   \n",
       "7008            M877T            0.2905      likely_benign  likely_pathogenic   \n",
       "375            K1879R            0.2235      likely_benign  likely_pathogenic   \n",
       "8508            R652K            0.5207          ambiguous  likely_pathogenic   \n",
       "5979           Q1030H            0.3505          ambiguous      likely_benign   \n",
       "7238            A843T            0.3338      likely_benign  likely_pathogenic   \n",
       "7964            Q734E            0.2651      likely_benign  likely_pathogenic   \n",
       "10763           T318P            0.4731          ambiguous  likely_pathogenic   \n",
       "3688           A1379S            0.3965          ambiguous  likely_pathogenic   \n",
       "7382            M822V            0.2755      likely_benign  likely_pathogenic   \n",
       "193            A1906G            0.3791          ambiguous  likely_pathogenic   \n",
       "7379            M822T            0.4654          ambiguous  likely_pathogenic   \n",
       "7172            M852R            0.4344          ambiguous  likely_pathogenic   \n",
       "8451            T660N            0.1591      likely_benign  likely_pathogenic   \n",
       "7630            R783G            0.2878      likely_benign  likely_pathogenic   \n",
       "7173            M852K            0.3783          ambiguous  likely_pathogenic   \n",
       "6689            E924G            0.4713          ambiguous  likely_pathogenic   \n",
       "8574            G641A            0.5164          ambiguous  likely_pathogenic   \n",
       "7246            S842G            0.2516      likely_benign  likely_pathogenic   \n",
       "7272            K837R            0.2118      likely_benign  likely_pathogenic   \n",
       "8782            Y609C            0.5154          ambiguous  likely_pathogenic   \n",
       "9960            M439V            0.4803          ambiguous  likely_pathogenic   \n",
       "6883            D896N            0.4739          ambiguous  likely_pathogenic   \n",
       "10247           D394E            0.2842      likely_benign  likely_pathogenic   \n",
       "6664            D928N            0.3180      likely_benign  likely_pathogenic   \n",
       "4911           R1193H            0.4463          ambiguous  likely_pathogenic   \n",
       "7202            E848G            0.5390          ambiguous  likely_pathogenic   \n",
       "6614            E935K            0.5593          ambiguous  likely_pathogenic   \n",
       "6524            E949K            0.5559          ambiguous  likely_pathogenic   \n",
       "7195            M849T            0.5624          ambiguous  likely_pathogenic   \n",
       "7085            K865R            0.1168      likely_benign  likely_pathogenic   \n",
       "7089            K865E            0.4285          ambiguous  likely_pathogenic   \n",
       "11538           I201T            0.2086      likely_benign  likely_pathogenic   \n",
       "6979            Q882E            0.3260      likely_benign  likely_pathogenic   \n",
       "7174            M852T            0.3544          ambiguous  likely_pathogenic   \n",
       "7061            R869C            0.3800          ambiguous  likely_pathogenic   \n",
       "6825            R904H            0.4928          ambiguous  likely_pathogenic   \n",
       "6650            E930Q            0.4718          ambiguous  likely_pathogenic   \n",
       "223            E1902Q            0.3783          ambiguous      likely_benign   \n",
       "7212            K847E            0.4053          ambiguous  likely_pathogenic   \n",
       "6896            E894G            0.3963          ambiguous  likely_pathogenic   \n",
       "6802            L908V            0.4203          ambiguous  likely_pathogenic   \n",
       "7947            I736T            0.3767          ambiguous  likely_pathogenic   \n",
       "4017           N1327K            0.8678  likely_pathogenic      likely_benign   \n",
       "7055            R870C            0.3476          ambiguous  likely_pathogenic   \n",
       "\n",
       "                                          ClinVar status  \n",
       "8260                 criteria provided, single submitter  \n",
       "11544                criteria provided, single submitter  \n",
       "8838                 criteria provided, single submitter  \n",
       "5822                 criteria provided, single submitter  \n",
       "10066                     no assertion criteria provided  \n",
       "5917                      no assertion criteria provided  \n",
       "7008                      no assertion criteria provided  \n",
       "375                  criteria provided, single submitter  \n",
       "8508                 criteria provided, single submitter  \n",
       "5979                 criteria provided, single submitter  \n",
       "7238                 criteria provided, single submitter  \n",
       "7964                 criteria provided, single submitter  \n",
       "10763                criteria provided, single submitter  \n",
       "3688                 criteria provided, single submitter  \n",
       "7382                 criteria provided, single submitter  \n",
       "193                  criteria provided, single submitter  \n",
       "7379                 criteria provided, single submitter  \n",
       "7172                 criteria provided, single submitter  \n",
       "8451   criteria provided, multiple submitters, no con...  \n",
       "7630                 criteria provided, single submitter  \n",
       "7173   criteria provided, multiple submitters, no con...  \n",
       "6689                 criteria provided, single submitter  \n",
       "8574   criteria provided, multiple submitters, no con...  \n",
       "7246                 criteria provided, single submitter  \n",
       "7272                 criteria provided, single submitter  \n",
       "8782   criteria provided, multiple submitters, no con...  \n",
       "9960   criteria provided, multiple submitters, no con...  \n",
       "6883                 criteria provided, single submitter  \n",
       "10247  criteria provided, multiple submitters, no con...  \n",
       "6664   criteria provided, multiple submitters, no con...  \n",
       "4911                            reviewed by expert panel  \n",
       "7202   criteria provided, multiple submitters, no con...  \n",
       "6614                 criteria provided, single submitter  \n",
       "6524                            reviewed by expert panel  \n",
       "7195   criteria provided, multiple submitters, no con...  \n",
       "7085   criteria provided, multiple submitters, no con...  \n",
       "7089                            reviewed by expert panel  \n",
       "11538  criteria provided, multiple submitters, no con...  \n",
       "6979   criteria provided, multiple submitters, no con...  \n",
       "7174   criteria provided, multiple submitters, no con...  \n",
       "7061   criteria provided, multiple submitters, no con...  \n",
       "6825                            reviewed by expert panel  \n",
       "6650   criteria provided, multiple submitters, no con...  \n",
       "223                             reviewed by expert panel  \n",
       "7212                            reviewed by expert panel  \n",
       "6896                            reviewed by expert panel  \n",
       "6802                            reviewed by expert panel  \n",
       "7947                            reviewed by expert panel  \n",
       "4017                            reviewed by expert panel  \n",
       "7055                            reviewed by expert panel  "
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Observe mismatches for both genes\n",
    "\n",
    "print(len(mismatch_dict[\"am_myh7\"]))\n",
    "mismatch_dict[\"am_myh7\"].head(50)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>genome</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>am_pathogenicity</th>\n",
       "      <th>am_class</th>\n",
       "      <th>ClinVar label</th>\n",
       "      <th>ClinVar status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11980</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47346217</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>K360N</td>\n",
       "      <td>0.4657</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8006</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47339740</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>V660M</td>\n",
       "      <td>0.2616</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10810</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47343029</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>F448S</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10110</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342697</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R502L</td>\n",
       "      <td>0.4558</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8084</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47339758</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R654G</td>\n",
       "      <td>0.2447</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10774</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47343021</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>E451Q</td>\n",
       "      <td>0.1996</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15890</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47351356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>T59A</td>\n",
       "      <td>0.0960</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11936</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47346207</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>A364T</td>\n",
       "      <td>0.1072</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8846</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47341991</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R597Q</td>\n",
       "      <td>0.1985</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11694</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47343571</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R382W</td>\n",
       "      <td>0.3417</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14648</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47350047</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>V158M</td>\n",
       "      <td>0.7314</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47332891</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R1138H</td>\n",
       "      <td>0.5375</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10206</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342719</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R495G</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13840</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47348541</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>V219L</td>\n",
       "      <td>0.4113</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5898</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47337534</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R820Q</td>\n",
       "      <td>0.3718</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342719</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R495W</td>\n",
       "      <td>0.1716</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10108</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342697</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R502Q</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10200</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342718</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>R495Q</td>\n",
       "      <td>0.0880</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13314</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47348424</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>E258K</td>\n",
       "      <td>0.1638</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9582</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47342578</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q14896</td>\n",
       "      <td>ENST00000545968.6</td>\n",
       "      <td>E542Q</td>\n",
       "      <td>0.1695</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS REF ALT genome uniprot_id      transcript_id  \\\n",
       "11980  chr11  47346217   C   G   hg38     Q14896  ENST00000545968.6   \n",
       "8006   chr11  47339740   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "10810  chr11  47343029   A   G   hg38     Q14896  ENST00000545968.6   \n",
       "10110  chr11  47342697   C   A   hg38     Q14896  ENST00000545968.6   \n",
       "8084   chr11  47339758   G   C   hg38     Q14896  ENST00000545968.6   \n",
       "10774  chr11  47343021   C   G   hg38     Q14896  ENST00000545968.6   \n",
       "15890  chr11  47351356   T   C   hg38     Q14896  ENST00000545968.6   \n",
       "11936  chr11  47346207   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "8846   chr11  47341991   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "11694  chr11  47343571   G   A   hg38     Q14896  ENST00000545968.6   \n",
       "14648  chr11  47350047   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "1786   chr11  47332891   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "10206  chr11  47342719   G   C   hg38     Q14896  ENST00000545968.6   \n",
       "13840  chr11  47348541   C   G   hg38     Q14896  ENST00000545968.6   \n",
       "5898   chr11  47337534   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "10208  chr11  47342719   G   A   hg38     Q14896  ENST00000545968.6   \n",
       "10108  chr11  47342697   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "10200  chr11  47342718   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "13314  chr11  47348424   C   T   hg38     Q14896  ENST00000545968.6   \n",
       "9582   chr11  47342578   C   G   hg38     Q14896  ENST00000545968.6   \n",
       "\n",
       "      protein_variant  am_pathogenicity           am_class      ClinVar label  \\\n",
       "11980           K360N            0.4657          ambiguous  likely_pathogenic   \n",
       "8006            V660M            0.2616      likely_benign  likely_pathogenic   \n",
       "10810           F448S            0.4113          ambiguous  likely_pathogenic   \n",
       "10110           R502L            0.4558          ambiguous  likely_pathogenic   \n",
       "8084            R654G            0.2447      likely_benign  likely_pathogenic   \n",
       "10774           E451Q            0.1996      likely_benign  likely_pathogenic   \n",
       "15890            T59A            0.0960      likely_benign  likely_pathogenic   \n",
       "11936           A364T            0.1072      likely_benign  likely_pathogenic   \n",
       "8846            R597Q            0.1985      likely_benign  likely_pathogenic   \n",
       "11694           R382W            0.3417          ambiguous      likely_benign   \n",
       "14648           V158M            0.7314  likely_pathogenic      likely_benign   \n",
       "1786           R1138H            0.5375          ambiguous      likely_benign   \n",
       "10206           R495G            0.1428      likely_benign  likely_pathogenic   \n",
       "13840           V219L            0.4113          ambiguous  likely_pathogenic   \n",
       "5898            R820Q            0.3718          ambiguous  likely_pathogenic   \n",
       "10208           R495W            0.1716      likely_benign  likely_pathogenic   \n",
       "10108           R502Q            0.1028      likely_benign  likely_pathogenic   \n",
       "10200           R495Q            0.0880      likely_benign  likely_pathogenic   \n",
       "13314           E258K            0.1638      likely_benign  likely_pathogenic   \n",
       "9582            E542Q            0.1695      likely_benign  likely_pathogenic   \n",
       "\n",
       "                                          ClinVar status  \n",
       "11980                criteria provided, single submitter  \n",
       "8006                 criteria provided, single submitter  \n",
       "10810                criteria provided, single submitter  \n",
       "10110                criteria provided, single submitter  \n",
       "8084                      no assertion criteria provided  \n",
       "10774                criteria provided, single submitter  \n",
       "15890                     no assertion criteria provided  \n",
       "11936  criteria provided, multiple submitters, no con...  \n",
       "8846   criteria provided, multiple submitters, no con...  \n",
       "11694  criteria provided, multiple submitters, no con...  \n",
       "14648  criteria provided, multiple submitters, no con...  \n",
       "1786   criteria provided, multiple submitters, no con...  \n",
       "10206  criteria provided, multiple submitters, no con...  \n",
       "13840  criteria provided, multiple submitters, no con...  \n",
       "5898   criteria provided, multiple submitters, no con...  \n",
       "10208  criteria provided, multiple submitters, no con...  \n",
       "10108  criteria provided, multiple submitters, no con...  \n",
       "10200  criteria provided, multiple submitters, no con...  \n",
       "13314  criteria provided, multiple submitters, no con...  \n",
       "9582   criteria provided, multiple submitters, no con...  "
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(mismatch_dict[\"am_mybpc3\"]))\n",
    "mismatch_dict[\"am_mybpc3\"].head(50)\n",
    "\n",
    "# mismatch_dict[\"mybpc3\"][\"ClinVar status\"].iloc[10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6011/2999807035.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"] = adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"].apply(lambda x: \"ambiguous\" if x == \"Uncertain\" else (\"likely_benign\" if \"Benign\" in x or \"benign\" in x else \"likely_pathogenic\"))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6011/2999807035.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  adj_eve_df[\"protein_variant\"] = adj_eve_df[\"wt_aa\"] + eve_df[\"position\"].apply(lambda x: str(x)) + eve_df[\"mt_aa\"]\n",
      "/tmp/ipykernel_6011/2999807035.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"] = adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"].apply(lambda x: \"ambiguous\" if x == \"Uncertain\" else (\"likely_benign\" if \"Benign\" in x or \"benign\" in x else \"likely_pathogenic\"))\n",
      "/tmp/ipykernel_6011/2999807035.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  adj_eve_df[\"protein_variant\"] = adj_eve_df[\"wt_aa\"] + eve_df[\"position\"].apply(lambda x: str(x)) + eve_df[\"mt_aa\"]\n"
     ]
    }
   ],
   "source": [
    "#Preprocess EVE dataframe\n",
    "\n",
    "\n",
    "eve_dfs = [eve_myh7_df, eve_mybpc3_df]\n",
    "adj_eve_dfs = []\n",
    "for eve_df in eve_dfs:\n",
    "    adj_eve_df = pd.DataFrame()\n",
    "\n",
    "    adj_eve_df = eve_df.dropna(subset=\"EVE_classes_75_pct_retained_ASM\")\n",
    "\n",
    "    #Standardize labels with those used above\n",
    "    adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"] = adj_eve_df[\"EVE_classes_75_pct_retained_ASM\"].apply(lambda x: \"ambiguous\" if x == \"Uncertain\" else (\"likely_benign\" if \"Benign\" in x or \"benign\" in x else \"likely_pathogenic\"))\n",
    "\n",
    "    #Add mutation label to match those of ClinVar\n",
    "    eve_df[\"protein_variant\"] = eve_df[\"wt_aa\"] + eve_df[\"position\"].apply(lambda x: str(x)) + eve_df[\"mt_aa\"]\n",
    "    adj_eve_df[\"protein_variant\"] = adj_eve_df[\"wt_aa\"] + eve_df[\"position\"].apply(lambda x: str(x)) + eve_df[\"mt_aa\"]\n",
    "\n",
    "    adj_eve_dfs.append(adj_eve_df)\n",
    "\n",
    "\n",
    "eve_myh7_df_adj = adj_eve_dfs[0]\n",
    "eve_mybpc3_df_adj = adj_eve_dfs[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n",
      "/tmp/ipykernel_6011/4046501550.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  mismatch_df = mismatch_df.append(matched_eve_pred)\n"
     ]
    }
   ],
   "source": [
    "#Find mismatches between EVE and ClinVar\n",
    "\n",
    "for cv_df, eve_df, gene in [(cv_myh7_df, eve_myh7_df_adj, \"eve_myh7\"), (cv_mybpc3_df, eve_mybpc3_df_adj, \"eve_mybpc3\")]:\n",
    "\n",
    "    mismatch_df = pd.DataFrame()\n",
    "    for i in range(len(cv_df)):\n",
    "        matched_eve_pred = eve_df[eve_df[\"protein_variant\"] == cv_df.iloc[i][\"Protein change\"]]\n",
    "\n",
    "        if not matched_eve_pred.empty:\n",
    "            matched_eve_pred = matched_eve_pred.iloc[0]\n",
    "\n",
    "            exp_class = cv_df.iloc[i][\"label\"]\n",
    "            status = cv_df.iloc[i][\"Review status\"]\n",
    "            pred_class = matched_eve_pred[\"EVE_classes_75_pct_retained_ASM\"]\n",
    "\n",
    "            if exp_class != \"ambiguous\":\n",
    "                if exp_class != pred_class:\n",
    "\n",
    "                    matched_eve_pred[\"ClinVar label\"] = exp_class\n",
    "                    matched_eve_pred[\"ClinVar status\"] = status\n",
    "\n",
    "                    # print(matched_eve_pred)\n",
    "                    # print(\"flow\")\n",
    "\n",
    "                    mismatch_df = mismatch_df.append(matched_eve_pred)\n",
    "    \n",
    "    mismatch_dict[gene] = mismatch_df[[\"protein_variant\", \"EVE_scores_ASM\", \"EVE_classes_75_pct_retained_ASM\", \"ClinVar label\", \"uncertainty_ASM\", \"ClinVar status\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "153\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>EVE_scores_ASM</th>\n",
       "      <th>EVE_classes_75_pct_retained_ASM</th>\n",
       "      <th>ClinVar label</th>\n",
       "      <th>uncertainty_ASM</th>\n",
       "      <th>ClinVar status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10720</th>\n",
       "      <td>E537A</td>\n",
       "      <td>0.613361</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.667221</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6033</th>\n",
       "      <td>L302Q</td>\n",
       "      <td>0.350133</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.647529</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13769</th>\n",
       "      <td>V689L</td>\n",
       "      <td>0.911706</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.298573</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3997</th>\n",
       "      <td>A200V</td>\n",
       "      <td>0.424342</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.681655</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12017</th>\n",
       "      <td>L601V</td>\n",
       "      <td>0.364945</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.656210</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12060</th>\n",
       "      <td>T604A</td>\n",
       "      <td>0.344009</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.643659</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35733</th>\n",
       "      <td>E1787Q</td>\n",
       "      <td>0.619193</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.664458</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8235</th>\n",
       "      <td>T412S</td>\n",
       "      <td>0.357886</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.652192</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4333</th>\n",
       "      <td>E217Q</td>\n",
       "      <td>0.622046</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.663054</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2954</th>\n",
       "      <td>S148R</td>\n",
       "      <td>0.525667</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.691829</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23974</th>\n",
       "      <td>S1199R</td>\n",
       "      <td>0.511561</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692880</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21090</th>\n",
       "      <td>L1055M</td>\n",
       "      <td>0.490499</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692967</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18233</th>\n",
       "      <td>K912Q</td>\n",
       "      <td>0.627301</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.660376</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17991</th>\n",
       "      <td>D900N</td>\n",
       "      <td>0.372541</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.660294</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14657</th>\n",
       "      <td>G733V</td>\n",
       "      <td>0.487092</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692814</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7149</th>\n",
       "      <td>H358L</td>\n",
       "      <td>0.567800</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.683925</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7331</th>\n",
       "      <td>K367N</td>\n",
       "      <td>0.617093</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.665469</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7679</th>\n",
       "      <td>S384Y</td>\n",
       "      <td>0.329737</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.633992</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8456</th>\n",
       "      <td>A423T</td>\n",
       "      <td>0.398227</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.672286</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3854</th>\n",
       "      <td>Q193R</td>\n",
       "      <td>0.421961</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.680917</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20794</th>\n",
       "      <td>Q1040R</td>\n",
       "      <td>0.289391</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.601605</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17536</th>\n",
       "      <td>M877T</td>\n",
       "      <td>0.167544</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.451970</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8829</th>\n",
       "      <td>R442L</td>\n",
       "      <td>0.389225</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.668400</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16508</th>\n",
       "      <td>N826K</td>\n",
       "      <td>0.414550</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.678472</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17773</th>\n",
       "      <td>L889Q</td>\n",
       "      <td>0.379435</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.663787</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37574</th>\n",
       "      <td>K1879R</td>\n",
       "      <td>0.423762</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.681477</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13028</th>\n",
       "      <td>R652K</td>\n",
       "      <td>0.102893</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.331393</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13313</th>\n",
       "      <td>H666Q</td>\n",
       "      <td>0.082942</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.285897</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3287</th>\n",
       "      <td>M165I</td>\n",
       "      <td>0.611709</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.667978</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15488</th>\n",
       "      <td>E775K</td>\n",
       "      <td>0.420954</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.680598</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12977</th>\n",
       "      <td>A649V</td>\n",
       "      <td>0.338544</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.640065</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20586</th>\n",
       "      <td>Q1030H</td>\n",
       "      <td>0.582270</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.679549</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5063</th>\n",
       "      <td>A254E</td>\n",
       "      <td>0.344068</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.643697</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10769</th>\n",
       "      <td>M539L</td>\n",
       "      <td>0.463032</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.690411</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31812</th>\n",
       "      <td>L1591P</td>\n",
       "      <td>0.347719</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.646023</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16856</th>\n",
       "      <td>A843T</td>\n",
       "      <td>0.429742</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.683242</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16669</th>\n",
       "      <td>F834L</td>\n",
       "      <td>0.359679</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.653233</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26988</th>\n",
       "      <td>E1350K</td>\n",
       "      <td>0.541925</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.689628</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18084</th>\n",
       "      <td>C905F</td>\n",
       "      <td>0.515993</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692636</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10470</th>\n",
       "      <td>I524M</td>\n",
       "      <td>0.509212</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692977</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17614</th>\n",
       "      <td>L881R</td>\n",
       "      <td>0.378882</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.663514</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11414</th>\n",
       "      <td>G571R</td>\n",
       "      <td>0.095210</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.314428</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6352</th>\n",
       "      <td>T318P</td>\n",
       "      <td>0.460104</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.689960</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27575</th>\n",
       "      <td>A1379S</td>\n",
       "      <td>0.524570</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.691939</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3365</th>\n",
       "      <td>R169G</td>\n",
       "      <td>0.409412</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.676644</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3375</th>\n",
       "      <td>R169S</td>\n",
       "      <td>0.420462</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.680441</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29003</th>\n",
       "      <td>K1451E</td>\n",
       "      <td>0.517359</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692544</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38485</th>\n",
       "      <td>R1925G</td>\n",
       "      <td>0.575644</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.681659</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31932</th>\n",
       "      <td>L1597P</td>\n",
       "      <td>0.630906</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.658472</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11726</th>\n",
       "      <td>D587H</td>\n",
       "      <td>0.307002</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.616681</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_variant  EVE_scores_ASM EVE_classes_75_pct_retained_ASM  \\\n",
       "10720           E537A        0.613361                       ambiguous   \n",
       "6033            L302Q        0.350133                   likely_benign   \n",
       "13769           V689L        0.911706               likely_pathogenic   \n",
       "3997            A200V        0.424342                       ambiguous   \n",
       "12017           L601V        0.364945                       ambiguous   \n",
       "12060           T604A        0.344009                   likely_benign   \n",
       "35733          E1787Q        0.619193                       ambiguous   \n",
       "8235            T412S        0.357886                   likely_benign   \n",
       "4333            E217Q        0.622046                       ambiguous   \n",
       "2954            S148R        0.525667                       ambiguous   \n",
       "23974          S1199R        0.511561                       ambiguous   \n",
       "21090          L1055M        0.490499                       ambiguous   \n",
       "18233           K912Q        0.627301                       ambiguous   \n",
       "17991           D900N        0.372541                       ambiguous   \n",
       "14657           G733V        0.487092                       ambiguous   \n",
       "7149            H358L        0.567800                       ambiguous   \n",
       "7331            K367N        0.617093                       ambiguous   \n",
       "7679            S384Y        0.329737                   likely_benign   \n",
       "8456            A423T        0.398227                       ambiguous   \n",
       "3854            Q193R        0.421961                       ambiguous   \n",
       "20794          Q1040R        0.289391                   likely_benign   \n",
       "17536           M877T        0.167544                   likely_benign   \n",
       "8829            R442L        0.389225                       ambiguous   \n",
       "16508           N826K        0.414550                       ambiguous   \n",
       "17773           L889Q        0.379435                       ambiguous   \n",
       "37574          K1879R        0.423762                       ambiguous   \n",
       "13028           R652K        0.102893                   likely_benign   \n",
       "13313           H666Q        0.082942                   likely_benign   \n",
       "3287            M165I        0.611709                       ambiguous   \n",
       "15488           E775K        0.420954                       ambiguous   \n",
       "12977           A649V        0.338544                   likely_benign   \n",
       "20586          Q1030H        0.582270                       ambiguous   \n",
       "5063            A254E        0.344068                   likely_benign   \n",
       "10769           M539L        0.463032                       ambiguous   \n",
       "31812          L1591P        0.347719                   likely_benign   \n",
       "16856           A843T        0.429742                       ambiguous   \n",
       "16669           F834L        0.359679                       ambiguous   \n",
       "26988          E1350K        0.541925                       ambiguous   \n",
       "18084           C905F        0.515993                       ambiguous   \n",
       "10470           I524M        0.509212                       ambiguous   \n",
       "17614           L881R        0.378882                       ambiguous   \n",
       "11414           G571R        0.095210                   likely_benign   \n",
       "6352            T318P        0.460104                       ambiguous   \n",
       "27575          A1379S        0.524570                       ambiguous   \n",
       "3365            R169G        0.409412                       ambiguous   \n",
       "3375            R169S        0.420462                       ambiguous   \n",
       "29003          K1451E        0.517359                       ambiguous   \n",
       "38485          R1925G        0.575644                       ambiguous   \n",
       "31932          L1597P        0.630906                       ambiguous   \n",
       "11726           D587H        0.307002                   likely_benign   \n",
       "\n",
       "           ClinVar label  uncertainty_ASM                       ClinVar status  \n",
       "10720  likely_pathogenic         0.667221  criteria provided, single submitter  \n",
       "6033   likely_pathogenic         0.647529  criteria provided, single submitter  \n",
       "13769      likely_benign         0.298573  criteria provided, single submitter  \n",
       "3997   likely_pathogenic         0.681655  criteria provided, single submitter  \n",
       "12017  likely_pathogenic         0.656210  criteria provided, single submitter  \n",
       "12060  likely_pathogenic         0.643659  criteria provided, single submitter  \n",
       "35733  likely_pathogenic         0.664458  criteria provided, single submitter  \n",
       "8235   likely_pathogenic         0.652192  criteria provided, single submitter  \n",
       "4333   likely_pathogenic         0.663054  criteria provided, single submitter  \n",
       "2954   likely_pathogenic         0.691829  criteria provided, single submitter  \n",
       "23974  likely_pathogenic         0.692880  criteria provided, single submitter  \n",
       "21090  likely_pathogenic         0.692967  criteria provided, single submitter  \n",
       "18233  likely_pathogenic         0.660376  criteria provided, single submitter  \n",
       "17991  likely_pathogenic         0.660294  criteria provided, single submitter  \n",
       "14657  likely_pathogenic         0.692814  criteria provided, single submitter  \n",
       "7149   likely_pathogenic         0.683925       no assertion criteria provided  \n",
       "7331   likely_pathogenic         0.665469       no assertion criteria provided  \n",
       "7679   likely_pathogenic         0.633992       no assertion criteria provided  \n",
       "8456   likely_pathogenic         0.672286       no assertion criteria provided  \n",
       "3854   likely_pathogenic         0.680917       no assertion criteria provided  \n",
       "20794  likely_pathogenic         0.601605       no assertion criteria provided  \n",
       "17536  likely_pathogenic         0.451970       no assertion criteria provided  \n",
       "8829   likely_pathogenic         0.668400  criteria provided, single submitter  \n",
       "16508  likely_pathogenic         0.678472  criteria provided, single submitter  \n",
       "17773  likely_pathogenic         0.663787  criteria provided, single submitter  \n",
       "37574  likely_pathogenic         0.681477  criteria provided, single submitter  \n",
       "13028  likely_pathogenic         0.331393  criteria provided, single submitter  \n",
       "13313  likely_pathogenic         0.285897       no assertion criteria provided  \n",
       "3287   likely_pathogenic         0.667978  criteria provided, single submitter  \n",
       "15488  likely_pathogenic         0.680598  criteria provided, single submitter  \n",
       "12977  likely_pathogenic         0.640065  criteria provided, single submitter  \n",
       "20586      likely_benign         0.679549  criteria provided, single submitter  \n",
       "5063   likely_pathogenic         0.643697  criteria provided, single submitter  \n",
       "10769  likely_pathogenic         0.690411  criteria provided, single submitter  \n",
       "31812  likely_pathogenic         0.646023  criteria provided, single submitter  \n",
       "16856  likely_pathogenic         0.683242  criteria provided, single submitter  \n",
       "16669  likely_pathogenic         0.653233  criteria provided, single submitter  \n",
       "26988  likely_pathogenic         0.689628  criteria provided, single submitter  \n",
       "18084  likely_pathogenic         0.692636  criteria provided, single submitter  \n",
       "10470  likely_pathogenic         0.692977       no assertion criteria provided  \n",
       "17614  likely_pathogenic         0.663514  criteria provided, single submitter  \n",
       "11414  likely_pathogenic         0.314428  criteria provided, single submitter  \n",
       "6352   likely_pathogenic         0.689960  criteria provided, single submitter  \n",
       "27575  likely_pathogenic         0.691939  criteria provided, single submitter  \n",
       "3365   likely_pathogenic         0.676644       no assertion criteria provided  \n",
       "3375   likely_pathogenic         0.680441  criteria provided, single submitter  \n",
       "29003  likely_pathogenic         0.692544       no assertion criteria provided  \n",
       "38485  likely_pathogenic         0.681659       no assertion criteria provided  \n",
       "31932  likely_pathogenic         0.658472  criteria provided, single submitter  \n",
       "11726  likely_pathogenic         0.616681  criteria provided, single submitter  "
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(mismatch_dict[\"eve_myh7\"]))\n",
    "mismatch_dict[\"eve_myh7\"].head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>EVE_scores_ASM</th>\n",
       "      <th>EVE_classes_75_pct_retained_ASM</th>\n",
       "      <th>ClinVar label</th>\n",
       "      <th>uncertainty_ASM</th>\n",
       "      <th>ClinVar status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13190</th>\n",
       "      <td>V660M</td>\n",
       "      <td>0.394654</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.670785</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22876</th>\n",
       "      <td>M1144T</td>\n",
       "      <td>0.432483</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.684002</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10029</th>\n",
       "      <td>R502L</td>\n",
       "      <td>0.616226</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.665881</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13065</th>\n",
       "      <td>R654G</td>\n",
       "      <td>0.124543</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.375881</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25264</th>\n",
       "      <td>C1264F</td>\n",
       "      <td>0.392013</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.669640</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15222</th>\n",
       "      <td>V762D</td>\n",
       "      <td>0.271664</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.584906</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7276</th>\n",
       "      <td>A364T</td>\n",
       "      <td>0.452626</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.688652</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20038</th>\n",
       "      <td>R1002W</td>\n",
       "      <td>0.677674</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.628610</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7638</th>\n",
       "      <td>R382W</td>\n",
       "      <td>0.466912</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.690956</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22746</th>\n",
       "      <td>R1138H</td>\n",
       "      <td>0.842475</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.435542</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3526</th>\n",
       "      <td>R177H</td>\n",
       "      <td>0.560917</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.685707</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9885</th>\n",
       "      <td>R495G</td>\n",
       "      <td>0.536363</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.690500</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4369</th>\n",
       "      <td>V219L</td>\n",
       "      <td>0.384258</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.666110</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16393</th>\n",
       "      <td>R820Q</td>\n",
       "      <td>0.512878</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692815</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10033</th>\n",
       "      <td>R502Q</td>\n",
       "      <td>0.396974</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.671766</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19943</th>\n",
       "      <td>Q998E</td>\n",
       "      <td>0.543869</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.689293</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17910</th>\n",
       "      <td>V896M</td>\n",
       "      <td>0.451316</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.688399</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>R495Q</td>\n",
       "      <td>0.548398</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.688455</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148</th>\n",
       "      <td>E258K</td>\n",
       "      <td>0.519496</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.692387</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_variant  EVE_scores_ASM EVE_classes_75_pct_retained_ASM  \\\n",
       "13190           V660M        0.394654                       ambiguous   \n",
       "22876          M1144T        0.432483                       ambiguous   \n",
       "10029           R502L        0.616226                       ambiguous   \n",
       "13065           R654G        0.124543                   likely_benign   \n",
       "25264          C1264F        0.392013                       ambiguous   \n",
       "15222           V762D        0.271664                   likely_benign   \n",
       "7276            A364T        0.452626                       ambiguous   \n",
       "20038          R1002W        0.677674               likely_pathogenic   \n",
       "7638            R382W        0.466912                       ambiguous   \n",
       "22746          R1138H        0.842475               likely_pathogenic   \n",
       "3526            R177H        0.560917                       ambiguous   \n",
       "9885            R495G        0.536363                       ambiguous   \n",
       "4369            V219L        0.384258                       ambiguous   \n",
       "16393           R820Q        0.512878                       ambiguous   \n",
       "10033           R502Q        0.396974                       ambiguous   \n",
       "19943           Q998E        0.543869                       ambiguous   \n",
       "17910           V896M        0.451316                       ambiguous   \n",
       "9893            R495Q        0.548398                       ambiguous   \n",
       "5148            E258K        0.519496                       ambiguous   \n",
       "\n",
       "           ClinVar label  uncertainty_ASM  \\\n",
       "13190  likely_pathogenic         0.670785   \n",
       "22876      likely_benign         0.684002   \n",
       "10029  likely_pathogenic         0.665881   \n",
       "13065  likely_pathogenic         0.375881   \n",
       "25264  likely_pathogenic         0.669640   \n",
       "15222  likely_pathogenic         0.584906   \n",
       "7276   likely_pathogenic         0.688652   \n",
       "20038      likely_benign         0.628610   \n",
       "7638       likely_benign         0.690956   \n",
       "22746      likely_benign         0.435542   \n",
       "3526       likely_benign         0.685707   \n",
       "9885   likely_pathogenic         0.690500   \n",
       "4369   likely_pathogenic         0.666110   \n",
       "16393  likely_pathogenic         0.692815   \n",
       "10033  likely_pathogenic         0.671766   \n",
       "19943      likely_benign         0.689293   \n",
       "17910      likely_benign         0.688399   \n",
       "9893   likely_pathogenic         0.688455   \n",
       "5148   likely_pathogenic         0.692387   \n",
       "\n",
       "                                          ClinVar status  \n",
       "13190                criteria provided, single submitter  \n",
       "22876                criteria provided, single submitter  \n",
       "10029                criteria provided, single submitter  \n",
       "13065                     no assertion criteria provided  \n",
       "25264                criteria provided, single submitter  \n",
       "15222  criteria provided, multiple submitters, no con...  \n",
       "7276   criteria provided, multiple submitters, no con...  \n",
       "20038  criteria provided, multiple submitters, no con...  \n",
       "7638   criteria provided, multiple submitters, no con...  \n",
       "22746  criteria provided, multiple submitters, no con...  \n",
       "3526   criteria provided, multiple submitters, no con...  \n",
       "9885   criteria provided, multiple submitters, no con...  \n",
       "4369   criteria provided, multiple submitters, no con...  \n",
       "16393  criteria provided, multiple submitters, no con...  \n",
       "10033  criteria provided, multiple submitters, no con...  \n",
       "19943  criteria provided, multiple submitters, no con...  \n",
       "17910  criteria provided, multiple submitters, no con...  \n",
       "9893   criteria provided, multiple submitters, no con...  \n",
       "5148   criteria provided, multiple submitters, no con...  "
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(mismatch_dict[\"eve_mybpc3\"]))\n",
    "mismatch_dict[\"eve_mybpc3\"].head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Put all variants with at least one mismatch in table\n",
    "\n",
    "#Start by standardizing column names\n",
    "\n",
    "eve_gene_dict = {\"eve_myh7\": eve_myh7_df, \"eve_mybpc3\": eve_mybpc3_df}\n",
    "am_gene_dict = {\"am_myh7\": am_myh7_df, \"am_mybpc3\": am_mybpc3_df}\n",
    "\n",
    "print(len(mismatch_dict))\n",
    "\n",
    "print(mismatch_dict.keys())\n",
    "\n",
    "for key in mismatch_dict.keys():\n",
    "    if \"am\" in key:\n",
    "        mismatch_dict[key].drop(columns=[\"CHROM\", \"POS\", \"REF\", \"ALT\", \"genome\", \"uniprot_id\", \"transcript_id\"], inplace=True)\n",
    "        mismatch_dict[key][\"EVE_scores_ASM\"] = eve_gene_dict[key.replace(\"am\", \"eve\")][eve_gene_dict[key.replace(\"am\", \"eve\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"EVE_scores_ASM\"].values\n",
    "        mismatch_dict[key][\"EVE_classes_75_pct_retained_ASM\"] = eve_gene_dict[key.replace(\"am\", \"eve\")][eve_gene_dict[key.replace(\"am\", \"eve\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"EVE_classes_75_pct_retained_ASM\"].values                      \n",
    "    else:\n",
    "        mismatch_dict[key][\"am_pathogenicity\"] = am_gene_dict[key.replace(\"eve\", \"am\")][am_gene_dict[key.replace(\"eve\", \"am\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"am_pathogenicity\"].values\n",
    "        mismatch_dict[key][\"am_class\"] = am_gene_dict[key.replace(\"eve\", \"am\")][am_gene_dict[key.replace(\"eve\", \"am\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"am_class\"].values\n",
    "\n",
    "    print(eve_gene_dict[key.replace(\"am\", \"eve\")][eve_gene_dict[key.replace(\"am\", \"eve\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"EVE_scores_ASM\"])\n",
    "    print(eve_gene_dict[key.replace(\"am\", \"eve\")][eve_gene_dict[key.replace(\"am\", \"eve\")][\"protein_variant\"].isin(mismatch_dict[key][\"protein_variant\"].values)][\"EVE_classes_75_pct_retained_ASM\"])\n",
    "    print(mismatch_dict[key])\n",
    "    break\n",
    "\n",
    "concat_df = pd.DataFrame()\n",
    "for key in mismatch_dict.keys():\n",
    "    concat_df = pd.concat([concat_df, mismatch_dict[key]])\n",
    "\n",
    "print(len(concat_df))\n",
    "\n",
    "concat_df.to_csv(path_or_buf=\"three_way_comp.csv\", index=False)\n",
    "concat_df.head(50)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ClinVar: 3732\n",
      "Mismatch Count: 61\n",
      "Match Count: 1924\n",
      "Newly Classified: 1642\n",
      "{'likely_pathogenic': {'true_pos': 271, 'false_pos': 5, 'false_neg': 54}, 'likely_benign_or_ambi': {'true_pos': 13, 'false_pos': 54, 'false_neg': 5}}\n",
      "PRECISION LIST FOR am_myh7: ['Precision for likely_pathogenic: 0.9818840579710145', 'Precision for likely_benign_or_ambi: 0.19402985074626866']\n",
      "RECALL LIST FOR am_myh7: ['Recall for likely_pathogenic: 0.8338461538461538', 'Recall for likely_benign_or_ambi: 0.7222222222222222']\n",
      "Number of ClinVar: 3036\n",
      "Mismatch Count: 20\n",
      "Match Count: 1149\n",
      "Newly Classified: 1102\n",
      "{'likely_pathogenic': {'true_pos': 22, 'false_pos': 1, 'false_neg': 17}, 'likely_benign_or_ambi': {'true_pos': 27, 'false_pos': 17, 'false_neg': 1}}\n",
      "PRECISION LIST FOR am_mybpc3: ['Precision for likely_pathogenic: 0.9565217391304348', 'Precision for likely_benign_or_ambi: 0.6136363636363636']\n",
      "RECALL LIST FOR am_mybpc3: ['Recall for likely_pathogenic: 0.5641025641025641', 'Recall for likely_benign_or_ambi: 0.9642857142857143']\n",
      "Number of ClinVar: 3732\n",
      "Mismatch Count: 153\n",
      "Match Count: 1819\n",
      "Newly Classified: 1630\n",
      "{'likely_pathogenic': {'true_pos': 181, 'false_pos': 2, 'false_neg': 143}, 'likely_benign_or_ambi': {'true_pos': 16, 'false_pos': 143, 'false_neg': 2}}\n",
      "PRECISION LIST FOR eve_myh7: ['Precision for likely_pathogenic: 0.9890710382513661', 'Precision for likely_benign_or_ambi: 0.10062893081761007']\n",
      "RECALL LIST FOR eve_myh7: ['Recall for likely_pathogenic: 0.558641975308642', 'Recall for likely_benign_or_ambi: 0.8888888888888888']\n",
      "Number of ClinVar: 3036\n",
      "Mismatch Count: 19\n",
      "Match Count: 928\n",
      "Newly Classified: 890\n",
      "{'likely_pathogenic': {'true_pos': 22, 'false_pos': 2, 'false_neg': 12}, 'likely_benign_or_ambi': {'true_pos': 21, 'false_pos': 12, 'false_neg': 2}}\n",
      "PRECISION LIST FOR eve_mybpc3: ['Precision for likely_pathogenic: 0.9166666666666666', 'Precision for likely_benign_or_ambi: 0.6363636363636364']\n",
      "RECALL LIST FOR eve_mybpc3: ['Recall for likely_pathogenic: 0.6470588235294118', 'Recall for likely_benign_or_ambi: 0.9130434782608695']\n",
      "{'am_myh7': 0.9692695214105793, 'am_mybpc3': 0.9828913601368692, 'eve_myh7': 0.9224137931034483, 'eve_mybpc3': 0.9799366420274551}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'\\nQUESTIONS\\n---Should we include the newly classified variants in the performacy metrics or not?\\n---Should the micro- or macro- average classified variants be used for analysis (or both)?\\n'"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Find accuracy of AlphaMissense and EVE for both each model-gene pair_scores = {}\n",
    "accuracy_scores = {}\n",
    "micro_avg_precision_scores = {}\n",
    "micro_avg_recall_scores = {}\n",
    "macro_avg_precision_scores = {}\n",
    "macro_avg_recall_scores = {}\n",
    "count = 0\n",
    "\n",
    "for cv_df, pred_df, model_gene_pair in [(cv_myh7_df, am_myh7_df, \"am_myh7\"), (cv_mybpc3_df, am_mybpc3_df, \"am_mybpc3\"),\n",
    "                           (cv_myh7_df, eve_myh7_df_adj, \"eve_myh7\"), (cv_mybpc3_df, eve_mybpc3_df_adj, \"eve_mybpc3\")]:\n",
    "\n",
    "    # mismatch_df = pd.DataFrame()\n",
    "    matched_count = 0\n",
    "    mismatch_count = 0\n",
    "    newly_class = 0\n",
    "\n",
    "    #Define dictionary to hold values needed to find the precision and recall for each model-gene pair\n",
    "    metrics_dict = {\"likely_pathogenic\" : {\"true_pos\": 0, \"false_pos\": 0, \"false_neg\": 0},\n",
    "                    \"likely_benign_or_ambi\" : {\"true_pos\": 0, \"false_pos\": 0, \"false_neg\": 0}}\n",
    "    print(\"Number of ClinVar: \" + str(len(cv_df)))\n",
    "    for i in range(len(cv_df)):\n",
    "        matched_pred = pred_df[pred_df[\"protein_variant\"] == cv_df.iloc[i][\"Protein change\"]]\n",
    "\n",
    "        #Drops frameshift mutations and NaNs\n",
    "\n",
    "        if not matched_pred.empty:\n",
    "            # Database may have duplicates, only consider one row\n",
    "            matched_pred = matched_pred.iloc[0]\n",
    "\n",
    "            #Segment to find accuracy\n",
    "            exp_class = cv_df.iloc[i][\"label\"]\n",
    "            if count <= 1:\n",
    "                pred_class = matched_pred[\"am_class\"]\n",
    "            else:\n",
    "                pred_class = matched_pred[\"EVE_classes_75_pct_retained_ASM\"]\n",
    "\n",
    "            # print(\"Expected Class :\" + str(exp_class))\n",
    "            # print(\"Predicted Class :\" + str(pred_class))\n",
    "            if exp_class != \"ambiguous\":\n",
    "\n",
    "                if exp_class == pred_class:\n",
    "                    matched_count += 1\n",
    "                else:\n",
    "                    mismatch_count += 1\n",
    "            else:\n",
    "\n",
    "                #New prediction made by model, mark is as positive\n",
    "                matched_count += 1\n",
    "                newly_class += 1\n",
    "            \n",
    "\n",
    "\n",
    "            #Segment to find metrics for class (to derive prec. and recall for each model-gene pair)\n",
    "            curr_class = exp_class\n",
    "            \n",
    "            if curr_class != \"ambiguous\":\n",
    "                \n",
    "                #Use combination key if class is likely_benign or ambiguous\n",
    "                if curr_class == \"likely_benign\":\n",
    "                    curr_class = \"likely_benign_or_ambi\"\n",
    "                \n",
    "                if pred_class == \"likely_benign\" or pred_class == \"ambiguous\":\n",
    "                    pred_class = \"likely_benign_or_ambi\"\n",
    "\n",
    "                #True positive if class is predicted correctly\n",
    "                if pred_class == curr_class:\n",
    "                    metrics_dict[curr_class][\"true_pos\"] += 1\n",
    "                \n",
    "                else:\n",
    "\n",
    "                    # False negative for this class\n",
    "                    metrics_dict[curr_class][\"false_neg\"] += 1\n",
    "\n",
    "                    #False positive for the predicted class\n",
    "                    metrics_dict[pred_class][\"false_pos\"] += 1\n",
    "\n",
    "    accuracy_scores[model_gene_pair] = matched_count/(mismatch_count+matched_count)\n",
    "    print(\"Mismatch Count: \" + str(mismatch_count))\n",
    "    print(\"Match Count: \" + str(matched_count))\n",
    "    print(\"Newly Classified: \" + str(newly_class))\n",
    "\n",
    "    print(metrics_dict)\n",
    "\n",
    "    precision_list = [f\"Precision for {x}: \" + str(metrics_dict[x][\"true_pos\"]/(metrics_dict[x][\"true_pos\"]+metrics_dict[x][\"false_pos\"])) for x in metrics_dict.keys()]\n",
    "    print(f\"PRECISION LIST FOR {model_gene_pair}: \" + str(precision_list))\n",
    "    # macro_avg_precision_scores[model_gene_pair] = sum(precision_list)/len(metrics_dict)\n",
    "    \n",
    "    recall_list = [f\"Recall for {x}: \" + str(metrics_dict[x][\"true_pos\"]/(metrics_dict[x][\"true_pos\"]+metrics_dict[x][\"false_neg\"])) for x in metrics_dict.keys()]\n",
    "    print(f\"RECALL LIST FOR {model_gene_pair}: \" + str(recall_list))\n",
    "\n",
    "    # macro_avg_recall_scores[model_gene_pair] = sum(recall_list)/len(metrics_dict)\n",
    "\n",
    "    overall_tp = sum([metrics_dict[x][\"true_pos\"] for x in metrics_dict.keys()])\n",
    "    overall_fp = sum([metrics_dict[x][\"false_pos\"] for x in metrics_dict.keys()])\n",
    "    overall_fn = sum([metrics_dict[x][\"false_neg\"] for x in metrics_dict.keys()])\n",
    "\n",
    "    micro_avg_precision_scores[model_gene_pair] = overall_tp/(overall_tp + overall_fp)\n",
    "    micro_avg_recall_scores[model_gene_pair] = overall_tp/(overall_tp + overall_fn)\n",
    "\n",
    "    count += 1\n",
    "print(accuracy_scores)\n",
    "# print(micro_avg_precision_scores)\n",
    "# print(micro_avg_recall_scores)\n",
    "# print(macro_avg_precision_scores)\n",
    "# print(macro_avg_recall_scores)\n",
    "\n",
    "\n",
    "'''\n",
    "QUESTIONS\n",
    "---Should we include the newly classified variants in the performacy metrics or not?\n",
    "---Should the micro- or macro- average classified variants be used for analysis (or both)?\n",
    "'''\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get precision and recall\n",
    "\n",
    "'''Micro-avergaing both of the statistics will probably give a better view of the performance because we want and idea of how well the models\n",
    " do on the variants as individuals more than each of the categories'''\n",
    "\n",
    "\n",
    "\n",
    "#Then write up a page (plain english) about findings\n",
    "# And here consider the level of evidence\n",
    "# AFter this,building a classifier\n",
    "\n",
    "# Relearn XGBoost and hyperparameter tuning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Q734E']\n"
     ]
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
